WreckEngine/Game.cpp

```diff
@@ -1,23 +1,38 @@
 #include "Game.h"
 #include <iostream>
 
-Game::Game(GLprogram prog) : shader(prog) { }
+Game::Game()
+    : currState(nullptr)
+    , drawDebug(true)
+{
+}
+
+Game::Game(GLprogram prog)
+    : currState(nullptr)
+    , shader(prog)
+    , drawDebug(true)
+{
+}
 
 void Game::addState(shared<State> s) { 
-	if (!states.size()) 
-		currState = s.get(); 
-	states.push_back(s); 
+    if (!states.size()) 
+        currState = s.get(); 
+    states.push_back(s);
 }
 
 void Game::update(double dt) {
-	currState->update(dt);
+    if (!currState) return;
```

Comment:
technically speaking, that's a nono. All games should be created with some default state, since the state is what holds all the entities. Just add one with no special handler function, nbd